src/GitHub.App/Services/NavigationService.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.TextManager.Interop;
+using IServiceProvider = System.IServiceProvider;
+using GitHub.Models;
+
+namespace GitHub.Services
+{
+    [Export(typeof(INavigationService))]
+    public class NavigationService : INavigationService
```

Comment:
I've actually had to abstract out the methods for opening a diff/file for the PR details view. I put them in [`IPullRequestEditorService`](https://github.com/github/VisualStudio/blob/feature/pr-reviews/src/GitHub.Exports.Reactive/Services/IPullRequestEditorService.cs) on the `feature/pr-reviews` branch.

Should we think about merging these two services?

Reply:
Yes, that makes sense.  No point making it a separate service unless it's being used in multiple places.